integration-tests/docker/test_env/test_env_builder.go

```diff
@@ -0,0 +1,122 @@
+package test_env
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-testing-framework/logwatch"
+	"github.com/smartcontractkit/chainlink/integration-tests/types/node"
+	"os"
+)
+
+type CLTestEnvBuilder struct {
+	reusable             bool
+	hasLogWatch          bool
+	hasGeth              bool
+	hasMockServer        bool
+	clNodesCount         int
+	externalAdapterCount int
+	customNodeCsaKeys    []string
+	defaultNodeCsaKeys   []string
+}
+
+func NewCLTestEnvBuilder() *CLTestEnvBuilder {
+	return &CLTestEnvBuilder{
+		externalAdapterCount: 1,
+	}
+}
+
+func (m *CLTestEnvBuilder) WithLogWatcher() *CLTestEnvBuilder {
+	m.hasLogWatch = true
+	return m
+}
+
+func (m *CLTestEnvBuilder) WithCLNodes(clNodesCount int) *CLTestEnvBuilder {
+	m.clNodesCount = clNodesCount
+	return m
+}
+
+func (m *CLTestEnvBuilder) WithGeth() *CLTestEnvBuilder {
+	m.hasGeth = true
+	return m
+}
+
+func (m *CLTestEnvBuilder) WithReusable() *CLTestEnvBuilder {
+	m.reusable = true
```

Comment:
@skudasov why we need reusable flag everywhere? If you use the existing container name, testcontainers-go lib will connect to the container. Please check how I do it in mercury 0.3